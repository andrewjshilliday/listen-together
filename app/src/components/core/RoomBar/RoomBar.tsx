import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { ApplicationState } from '../../../store';
import styled from 'styled-components';

const RoomBar: React.FC = (props: any) => {
  const roomId = useSelector((state: ApplicationState) => state.room.roomId);
  
  return (
    <Link to={`/room/${roomId}`}>
      <RoomBarContainer>
        <i className="fas fa-door-open fa-2x"></i>
      </RoomBarContainer>
    </Link>
  );
}

export default RoomBar;


const RoomBarContainer = styled.div`
  background: var(--background-darker);
  border-right: 1px solid var(--on-background);
  height: calc(100vh - 175px);
  width: 50px;
  overflow: hidden;
  display: flex;
  align-items: center;
  justify-content: center;
`;
