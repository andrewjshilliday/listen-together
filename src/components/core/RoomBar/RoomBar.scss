.room-bar {
  background: var(--background-darker);
  border-right: 1px solid black;
  height: calc(100vh - 175px);
  width: 50px;
  overflow: hidden;
  display: flex;
  align-items: center;
  justify-content: center;
}
