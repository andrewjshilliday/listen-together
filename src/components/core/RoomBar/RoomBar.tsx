import React from 'react';
import { Link } from 'react-router-dom';
import { useRoom } from '../../providers'
import './RoomBar.scss';

const RoomBar: React.FC = (props: any) => {
  const roomProvider = useRoom();
  
  return (
    <Link to={`/room/${roomProvider.roomId}`}>
      <div className="room-bar">
        <i className="fas fa-door-open fa-2x"></i>
      </div>
    </Link>
  );
}

export default RoomBar;
