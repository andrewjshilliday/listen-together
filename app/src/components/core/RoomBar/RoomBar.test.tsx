import React from 'react';
import { StaticRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RoomBar from './RoomBar';

jest.mock('../../providers/RoomProvider', () => require('../../../__mocks__/providers/RoomProvider.mock').default);

it('renders correctly', () => {
  const tree = renderer.create(
    <StaticRouter location=".">
      <RoomBar />
    </StaticRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
