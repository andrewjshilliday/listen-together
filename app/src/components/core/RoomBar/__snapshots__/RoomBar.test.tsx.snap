// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/room/TEST1"
  onClick={[Function]}
>
  <div
    className="sc-AxjAm cgdwdU"
  >
    <i
      className="fas fa-door-open fa-2x"
    />
  </div>
</a>
`;
